Kernel/Devices/BlockDevice.h

```diff
@@ -68,6 +68,8 @@ class BlockDevice : public Device {
         : Device(major, minor)
         , m_block_size(block_size)
     {
+        // 512 is the minimum sector size in most block devices
+        VERIFY(m_block_size >= 512);
```

Comment:
Any particular reason this needs to be here? If we have no code that needs this, should this be a (pointless) requirement?

Reply:
I spent/wasted a lot of time because initially as I was passing a wrong `m_block_size` which lead to a memory corruption error. I thought it would be a good check to add to avoid future misery. 
Anyway, almost all storage devices has 512 bytes as a minimum block size, that is why added this check.